src/wp-includes/blocks.php

```diff
@@ -513,6 +514,39 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		}
 	}
 
+	if ( ! empty( $metadata['blockHooks'] ) ) {
+		/**
+		 * Map camelCased position string (from block.json) to snake_cased block type position.
+		 *
+		 * @var array
+		 */
+		$position_mappings = array(
+			'before'     => 'before',
+			'after'      => 'after',
+			'firstChild' => 'first_child',
+			'lastChild'  => 'last_child',
+		);
+
+		$settings['block_hooks'] = array();
+		foreach ( $metadata['blockHooks'] as $anchor_block_name => $position ) {
+			// Avoid infinite recursion (hooking to itself).
+			if ( $metadata['name'] === $anchor_block_name ) {
+				_doing_it_wrong(
+					__METHOD__,
+					__( 'Cannot hook block to itself.', 'gutenberg' ),
```

Comment:
We should remove the 'gutenberg' text domain.

Reply:
```suggestion
					__( 'Cannot hook block to itself.' ),
```

Reply:
I'm on it 👍🏻

Reply:
https://github.com/WordPress/wordpress-develop/pull/5218 - PR is ready.

Reply:
I committed with https://core.trac.wordpress.org/changeset/56592 as there is still ongoing discussion in the PR.